### PR TITLE
Use bytes instead of MB for memory limits

### LINKED ISSUE #438: Make ResourceLimiter memory representation consistent
Our API accepts memory limits in gigabytes, kubernetes keeps memory in bytes and we store it internally in megabytes. We should decide on one approach (bytes I think?) and update the code accordingly.

This issue is a follow up after discussion with @mwielgus on #429.

cc: @aleksandra-malinowska @krzysztof-jastrzebski

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Fixes #438.

### CODE REVIEW

File: cluster-autoscaler/config/const.go [hunk 1]
```diff
@@ -17,8 +17,15 @@ limitations under the License.
...
+
 	// DefaultMaxClusterCores is the default maximum number of cores in the cluster.
 	DefaultMaxClusterCores = 5000 * 64
 	// DefaultMaxClusterMemory is the default maximum number of gigabytes of memory in cluster.
 	DefaultMaxClusterMemory = 5000 * 64 * 20
+
+	// Useful universal constants.
+
+	// Gigabyte is 2^30 bytes.
+	Gigabyte = 1024 * 1024 * 1024
```
Comment (losipiuk) [inline cluster-autoscaler/config/const.go]: The fact this constant is in `config` module makes it look weird when it use used. Maybe consider creating separate module (consts/units?) for this one?

Comment (aleksandra-malinowska) [reply cluster-autoscaler/config/const.go]: I kind of agree, but I'd rather not add another module just for this constant

Comment (losipiuk) [reply cluster-autoscaler/config/const.go]: What is expensive about modules?

Comment (aleksandra-malinowska) [reply cluster-autoscaler/config/const.go]: Complexity. I'll move it since this is cleaner, it just seems like a bit of an overkill

Comment (aleksandra-malinowska) [reply cluster-autoscaler/config/const.go]: Done

Comment (losipiuk) [reply cluster-autoscaler/config/const.go]: Thanks. Can we rename `MB` to `Megabyte` and move it there too? Maybe there is some `KB` also? :)

File: cluster-autoscaler/utils/drain/drain.go [hunk 2]
```diff
@@ -31,9 +31,7 @@ import (
 const (
 	// PodDeletionTimeout - time after which a pod to be deleted is not included in the list of pods for drain.
 	PodDeletionTimeout = 12 * time.Minute
-)
 
-const (
```
Comment (kgolab) [inline cluster-autoscaler/utils/drain/drain.go]: This way we're mixing two distinct constant types together - I'd rather keep them separate.

Comment (aleksandra-malinowska) [reply cluster-autoscaler/utils/drain/drain.go]: Reverted

File: cluster-autoscaler/config/const.go (hunk 1 above)
Comment (aleksandra-malinowska) [reply cluster-autoscaler/config/const.go]: Do you mean MB from unit tests? If so, we don't use that in non-test code, so I'd rather keep it contained to test files

Comment (losipiuk) [reply cluster-autoscaler/config/const.go]: Yeah - I meant those. But I did not notice those are test only. If so leave it that way :)

**Comment by kgolab** [comment]
Fine for me.
The only nit is that we're using MB in test (for example 1300 MB) which are not possible using command-line arguments but it's not part of the change.

**Pull request merged**